src/main/java/net/fabricmc/loom/configuration/CompileConfiguration.java

```diff
@@ -157,8 +157,16 @@ private synchronized void setupMinecraft(ConfigContext configContext) throws Exc
 
 		var jarConfiguration = extension.getMinecraftJarConfiguration().get();
 
-		if (jarConfiguration == MinecraftJarConfiguration.MERGED && !metadataProvider.getVersionMeta().isVersionOrNewer(Constants.RELEASE_TIME_1_3)) {
-			jarConfiguration = MinecraftJarConfiguration.LEGACY_MERGED;
+		if (jarConfiguration == MinecraftJarConfiguration.MERGED) {
+			// if no configuration is selected by the user, attempt to select one
+			// based on the mc version and which sides are present for it
+			if (!metadataProvider.getVersionMeta().downloads().containsKey("server")) {
+				jarConfiguration = MinecraftJarConfiguration.CLIENT_ONLY;
+			} else if (!metadataProvider.getVersionMeta().downloads().containsKey("client")) {
+				jarConfiguration = MinecraftJarConfiguration.SERVER_ONLY;
+			} else if (!metadataProvider.getVersionMeta().isVersionOrNewer(Constants.RELEASE_TIME_1_3)) {
+				jarConfiguration = MinecraftJarConfiguration.LEGACY_MERGED;
```

Comment:
I dont think we should fall back to this, as this assumes that you have 3rd party intermediary mappings.

Reply:
it is the current behavior already so I kept it

Reply:
Oh, I see yeah that makes sense, as its already requested to be merged. thats fine.